### PR TITLE
feat(request_forwarder): add metric for external response times

### LINKED ISSUE #229: Add metric: time request sent by coco returns

### PR DESCRIPTION
**Pull request by nritsche**
Adds a histogram metric that observes the response times for requests
coco sends to external endpoints.

Closes #229

### CODE REVIEW

File: coco/request_forwarder.py [hunk 1]
```diff
@@ -265,14 +270,22 @@ async def _request(self, session, method, host, endpoint, request, params):
...
-                endpoint=endpoint, host=hostname, port=port, status="0"
+                endpoint=endpoint, host=hostname, port=port, status=status
             ).inc()
             return host, (str(e), 0)
+        finally:
+            response_time = time.time() - start_time
+            self.response_time.labels(
+                endpoint=endpoint, host=hostname, port=port
+            ).observe(response_time)
+            self.call_counter.labels(
```
Comment (anjakefala) [inline coco/request_forwarder.py:286]: If there is an exception, `call_counter` gets incremented twice?

Comment (anjakefala) [reply coco/request_forwarder.py:286]: Oh nvm, in the `except` blocks there is a `return`. So if there is an `exception`, we will get one `call_counter`, and no `response_time` (which makes sense. Okay.

Comment (nritsche) [reply coco/request_forwarder.py:286]: No, I think regardless of the `return`, `finally` will get executed, so we should  remove the `call_counter` from there.

Comment (anjakefala) [reply coco/request_forwarder.py:286]:
```
 def func():
     try:
         2 + 'hello'
     except:
         print('1')
         return
     finally:
         print('2')
 func()    
 ```
 ```
 anjakefala@chime:~$ python3 test.py 
1
2
```

You are right

Comment (anjakefala) [reply coco/request_forwarder.py:286]: Or move it to within the `try` block, after the statement that would normally trigger the exception.

**Review by anjakefala — CHANGES_REQUESTED** [review]
Change requested: move the `call_counter` so it does not get triggered twice, if there is an exception.

**Review by anjakefala — APPROVED** [review]

**Pull request merged**